src/components/VariableCreateModal.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <p-modal v-model:showModal="internalValue" :title="localization.info.newVariable">
+    <p-form @submit="submit">
+      <p-content>
+        <p-label :label="localization.info.name" :state="nameState" :message="errors.name">
+          <p-text-input v-model="name" :state="nameState" />
+        </p-label>
+
+        <p-label :label="localization.info.value" :state="valueState" :message="errors.value">
+          <p-text-input v-model="value" :state="valueState" />
+        </p-label>
+
+        <p-label :label="localization.info.tags" :state="tagsState" :message="errors.tags">
+          <p-tag-input v-model="tags" :state="tagsState" />
+        </p-label>
+      </p-content>
+    </p-form>
+
+    <template #actions>
+      <p-button :loading="isSubmitting" @click="submit">
+        Create
+      </p-button>
+    </template>
+    <template #cancel>
+      <p-button inset @click="internalValue = false">
+        Close
+      </p-button>
+    </template>
+  </p-modal>
+</template>
+
+<script lang="ts" setup>
+  import { showToast } from '@prefecthq/prefect-design'
+  import { useField } from 'vee-validate'
+  import { computed } from 'vue'
+  import { useWorkspaceApi, useForm } from '@/compositions'
+  import { localization } from '@/localization'
+  import { Variable } from '@/models'
+  import { fieldRules, isRequired, isValidIf } from '@/utilities'
+
+  const props = defineProps<{
+    showModal: boolean,
+  }>()
+
+  const emits = defineEmits<{
+    (event: 'update:showModal', value: boolean): void,
+  }>()
+
+  const internalValue = computed({
+    get() {
+      return props.showModal
+    },
+    set(value: boolean): void {
+      emits('update:showModal', value)
+    },
+  })
+
+  const api = useWorkspaceApi()
+
+  const isUnique = isValidIf(async () => {
+    const variable = await api.variables.getVariableByName(name.value)
+    return !variable
+  })
+  const nameRules = fieldRules(localization.info.name, isRequired, [isUnique, localization.error.variableAlreadyExists])
+  const valueRules = fieldRules(localization.info.value, isRequired)
+
+  const { handleSubmit, isSubmitting, errors } = useForm<Variable>()
+  const { value: name, meta: nameState } = useField<string>('name', nameRules)
+  const { value: value, meta: valueState } = useField<string>('value', valueRules)
+  const { value: tags, meta: tagsState } = useField<string[]>('tags')
+
+  const submit = handleSubmit(async (values) => {
```

Comment:
Can we not introduce more vee validate? I'd like to use the useValidation and useValidationObserver compositions

Reply:
You know it's been a minute since I was creating forms and I forgot which version I was using - yeah definitely, I'll update that